lib/pages/infoPages/vendorPage.dart

```diff
@@ -26,28 +38,63 @@ class _VendorPageWidgetState extends State<VendorPageWidget> {
   FirebaseUser user;
   DatabaseReference _userRef;
 
+  //Location variables
+  final _locationService = Geolocator();
+  Position currentLocation;
+
   //Declare contextual variables
   AppState appState;
   ThemeData theme;
 
   bool following = false;
 
+  Deals deals = Deals();
+  DatabaseReference _dealsRef;
+
   @override
   void initState() {
     super.initState();
+    currentLocation = widget.location;
     initialize();
   }
 
   void initialize() async {
     user = await _auth.currentUser();
     _userRef = FirebaseDatabase().reference().child("Users").child(user.uid);
+    _dealsRef = FirebaseDatabase().reference().child("Deals");
     _userRef.child("following").onValue.listen((data){
       if (data.snapshot != null){
         setState(() {
           following = data.snapshot.value[widget.vendor.key]?? false;
         });        
       }
     });
+    try {
+      var serviceStatus = await _locationService.checkGeolocationPermissionStatus();
+      if (serviceStatus == GeolocationStatus.granted) {//we should have permission but maybe they turned it off since they got here
+        _locationService.getPositionStream(LocationOptions(accuracy: LocationAccuracy.best)).listen((Position result) async {
+          if (this.mounted){
+            setState(() {
+              currentLocation = result;
+            });
+          }
+        });
+      }
+    } on PlatformException catch (e) {
+      print(e.message);
+    }
+    _dealsRef.orderByChild("vendor_id").equalTo(widget.vendor.key).onValue.listen((dealEvent) {
+      if (this.mounted && dealEvent.snapshot != null){
+        Map<String, dynamic> dealDataMap = new Map<String, dynamic>.from(dealEvent.snapshot.value);
+        setState(() {
+          dealDataMap.forEach((key,data){
+            Deal newDeal = new Deal.fromMap(key, data, widget.vendor, user.uid);
+            // newDeal.favorited = favorites.containsKey(newDeal.key);
+            deals.addDeal(newDeal);
```

Comment:
I think this logic should be moved outside of the setState. Shouldn't mutate the state, setState should receive a new Deals object